zerver/lib/message.py

```diff
@@ -1220,21 +1220,23 @@ def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: List[i
 
     for group in user_groups:
         can_mention_group = group.can_mention_group
-        can_mention_group_name = can_mention_group.named_user_group.name
+        if (
+            hasattr(can_mention_group, "named_user_group")
+            and can_mention_group.named_user_group.name == SystemGroups.EVERYONE
+        ):
+            continue
         if sender_is_system_bot:
-            if can_mention_group_name == SystemGroups.EVERYONE:
-                continue
             raise JsonableError(
-                _(
-                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of '{can_mention_group_name}' to mention this group."
-                ).format(user_group_name=group.name, can_mention_group_name=can_mention_group_name)
+                _("You are not allowed to mention user group '{user_group_name}'.").format(
+                    user_group_name=group.name
```

Comment:
I don't think we need special code for that case; it's a useless configuration, but not a problematic one.... and might even be a useful intermediate state to keep track of some individual permissions when temporarily changing the current permission value to `everyone` with intent to restrict it again later.

Reply:
Ok, so do we want to allow system bots to mention the group in case the setting is set to `{direct_subgroups: [everyone_group.id], direct_members: [admin.id]}`. We do not allow currently.